File: A2/A2-016/A2-016.py
PRIZE = [[100000, 200, 100, 0],
         [1000000, 2000, 1000, 20]]
# Input lottery and its result
lottery = input().strip()
result = input().strip()

# Check letter part
type01 = int(lottery[0] == result[0])
# Check number part
type02 = 3
if(lottery[2:] == result[2:]):
    type02 = 0
elif(lottery[4:] == result[4:]):
    type02 = 1
elif(lottery[5:] == result[5:]):
    type02 = 2

# Output
print(PRIZE[type01][type02])

File: A2/A2-016/A2-016.cpp
#include<iostream>
using namespace std;

const int PRIZE[2][4] = {{100000, 200, 100, 0},
                         {1000000, 2000, 1000, 20}};
int type, cnt = 0;
char lotteryChar, resultChar;
string lotteryNum, resultNum;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input lottery and its result
    cin >> lotteryChar >> lotteryNum;
    cin >> resultChar >> resultNum;

    // Check letter part
    type = (lotteryChar == resultChar);
    // Check number part
    for(int i = 4; i >= 0; i--)
    {
        if(lotteryNum[i] == resultNum[i]) { cnt++; }
        else { break; }
    }
    
    // Output
    if(cnt == 5) { cout << PRIZE[type][0]; }
    else if(cnt >= 3) { cout << PRIZE[type][1]; }
    else if(cnt == 2) { cout << PRIZE[type][2]; }
    else { cout << PRIZE[type][3]; }
    return 0;
}